### DISCUSSION TITLE
DXF issue with ACAD input

### DISCUSSION THREAD

**Discussion by FlorianKrenn** [mozman/ezdxf#1016] [Issues] [answered]
Dear all,

when reading a dxf (generated from ACAD). adding entities (just circles and text) and then saving the same, it becomes not legible for ACAD anymore.
Also just reading the file and then saving it in ezdxf will give a problem with ACAD program as well as AutoDesk Online viewer).
Interestingly LibreCad can read all the files without issues.

Thanks in advance.

**Comment by mozman** [comment]
Not enough information to help ...

- I need your script and the code should be executable out of the box.
- I need the DXF file to reproduce the problem.
- Upload DXF files in a zip-archive.
- What operating system are you using?
- Which version of `ezdxf` are you using:
    - Type `ezdxf -V` on the command line
    - or `py -c "import ezdxf; ezdxf.print_config()"`
    - or `python3 -c "import ezdxf; ezdxf.print_config()"`

**Comment by FlorianKrenn** [comment]
Thanks for the quick reply.
Will send you once on the mac again.

Best regards / Mit freundlichen Gruessen
Florian Krenn
----------------------------------------------------------------------
Dr. Florian Krenn
Managing Director
Mobile: +91 98 10092339

GEOCONSULT India Pvt.Ltd. CIN No : U74220DL2007PTC167317


Corporate Office:

04B106, WeWork

Platina Tower, MG Road

Near Sikandarpur Metro Station

Sector 28, Gurugram, HR 122 001

INDIA

Disclaimer: "All Disclaimer Clauses as applicable to electronic communications, under all the relevant Acts, are applicable to the contents of this email and its attachments"
________________________________

**Comment by FlorianKrenn** [comment]
To the questions –

  *   I am using a MacBook, MacOS 14.3
  *   ezdxf 1.1.0
  *   anaconda for Mac

Best regards / Mit freundlichen Gruessen

Florian Krenn

----------------------------------------------------------------------
Dr. Florian Krenn
Managing Director
Mobile: +91 98 10092339

GEOCONSULT India Pvt.Ltd.
CIN No: U74220DL2007PTC167317

Corporate Office:
04B106, WeWork
Platina Tower, MG Road
Near Sikandarpur Metro Station
Sector 28, Gurugram, HR 122 001
INDIA

[signature_2855349985]

Disclaimer: "All Disclaimer Clauses as applicable to electronic communications, under all the relevant Acts, are applicable to the contents of this email and its attachments"

From: Manfred Moitzi ***@***.***>
Date: Thursday, 25 January 2024 at 17:22
To: mozman/ezdxf ***@***.***>
Cc: Dr. Florian Krenn ***@***.***>, Author ***@***.***>
Subject: Re: [mozman/ezdxf] DXF issue with ACAD input (Discussion #1016)

CAUTION: This message was sent from outside of Geoconsult India. Please do not click links or open attachments unless you recognize the source of this email and know the content is safe. Please report all suspicious emails to ***@***.******@***.***>" as an attachment.

Not enough information to help ...
·         I need your script and the code should be executable out of the box.
·         I need the DXF file to reproduce the problem.
·         Upload DXF files in a zip-archive.
·         What operating system are you using?
·         Which version of ezdxf are you using:
o    Type ezdxf -V on the command line
o    or py -c "import ezdxf; ezdxf.print_config()"
o    or python3 -c "import ezdxf; ezdxf.print_config()"

**Reply by mozman** [reply]
Still not enough information to help ...

- I need your script and the code should be executable out of the box.
- I need the DXF file to reproduce the problem.
- Upload DXF files in a zip-archive.

The latest released `ezdxf` version is 1.1.4 - install via `pip` Anacoda packages are often outdated.

**Reply by FlorianKrenn** [reply]
[Script_dxf-files.zip](https://github.com/mozman/ezdxf/files/14071330/Script_dxf-files.zip)

the files ...

**Comment by FlorianKrenn** [comment]
Thanks for the tip to upgrade, unfortunately it did not help …

Files are attached …

Best regards / Mit freundlichen Gruessen

Florian Krenn

----------------------------------------------------------------------
Dr. Florian Krenn
Managing Director
Mobile: +91 98 10092339

GEOCONSULT India Pvt.Ltd.
CIN No: U74220DL2007PTC167317

Corporate Office:
04B106, WeWork
Platina Tower, MG Road
Near Sikandarpur Metro Station
Sector 28, Gurugram, HR 122 001
INDIA

[signature_768759089]

Disclaimer: "All Disclaimer Clauses as applicable to electronic communications, under all the relevant Acts, are applicable to the contents of this email and its attachments"

From: Manfred Moitzi ***@***.***>
Date: Friday, 26 January 2024 at 15:02
To: mozman/ezdxf ***@***.***>
Cc: Dr. Florian Krenn ***@***.***>, Author ***@***.***>
Subject: Re: [mozman/ezdxf] DXF issue with ACAD input (Discussion #1016)

CAUTION: This message was sent from outside of Geoconsult India. Please do not click links or open attachments unless you recognize the source of this email and know the content is safe. Please report all suspicious emails to ***@***.******@***.***>" as an attachment.

Still not enough information to help ...
·         I need your script and the code should be executable out of the box.
·         I need the DXF file to reproduce the problem.
·         Upload DXF files in a zip-archive.

The latest released ezdxf version is 1.1.4 - install via pip Anacoda packages are often outdated.

**Comment by mozman** [comment, marked as answer]
The default linetype is `CONTINUOUS` not `SOLID` and `SOLID` does not exist.

Error message from DWG Trueview 2024:
```
Error: Undefined line type SOLID in LayerTableRecord GrHo_4m_TXT
Invalid or incomplete DXF input -- drawing discarded.
Press ENTER to continue:
```

The solution is simple: duplicate  `CONTINUOUS` as `SOLID`
```python
doc.linetypes.duplicate_entry("CONTINUOUS", "SOLID")
```
BTW: BricsCAD is not that picky and opens the DXF file with missing linetype definitions

**Reply by mozman** [reply]
I extended the `Auditor` to fix non-existing linetypes in layers.  The auditing process is time consuming and therefor not called automatically, you run `doc.validate()` before exporting the DXF file:

```python
doc.validate()
doc.saveas("4m_holesCStxt.dxf")
```

Without creating the `SOLID` entry you get this output:
```
1 issues fixed.

   1. Issue [100] fixed in LAYER(#338).
   Replaced undefined linetype SOLID in layer 'GrHo_4m_TXT' by CONTINUOUS

No unrecoverable errors found.
```

**Comment by FlorianKrenn** [comment]
Dear Manfred Moitzi

Thanks a lot.
And please excuse my taking your time … that mistake on my end I completely overlooked (was not even on my radar).

I checked on my end – yes, that was it.

Thanks for your efforts and I really appreciate ezdxf ( using it for some “sketching”), it is a great tool !

Best regards / Mit freundlichen Gruessen

Florian Krenn

----------------------------------------------------------------------
Dr. Florian Krenn
Managing Director
Mobile: +91 98 10092339

GEOCONSULT India Pvt.Ltd.
CIN No: U74220DL2007PTC167317

Corporate Office:
04B106, WeWork
Platina Tower, MG Road
Near Sikandarpur Metro Station
Sector 28, Gurugram, HR 122 001
INDIA

[signature_3623775734]

Disclaimer: "All Disclaimer Clauses as applicable to electronic communications, under all the relevant Acts, are applicable to the contents of this email and its attachments"

From: Manfred Moitzi ***@***.***>
Date: Saturday, 27 January 2024 at 15:18
To: mozman/ezdxf ***@***.***>
Cc: Dr. Florian Krenn ***@***.***>, Author ***@***.***>
Subject: Re: [mozman/ezdxf] DXF issue with ACAD input (Discussion #1016)

CAUTION: This message was sent from outside of Geoconsult India. Please do not click links or open attachments unless you recognize the source of this email and know the content is safe. Please report all suspicious emails to ***@***.******@***.***>" as an attachment.

I extended the Auditor to fix non-existing linetypes in layers. The auditing process is time consuming and therefor not called automatically, you run doc.validate() before exporting the DXF file:

doc.validate()

doc.saveas("4m_holesCStxt.dxf")

Without creating the SOLID entry you get this output:

1 issues fixed.



   1. Issue [100] fixed in LAYER(#338).

   Replaced undefined linetype SOLID in layer 'GrHo_4m_TXT' by CONTINUOUS



No unrecoverable errors found.
